src/compiler/crystal/tools/doc/constant.cr

```diff
@@ -24,4 +25,12 @@ class Crystal::Doc::Constant
   def formatted_value
     Highlighter.highlight value.to_s
   end
+
+  def to_json(io)
+    {
+      :name => name,
+      :value => value.to_s,
+      :doc => doc
+    }.to_json(io)
```

Comment:
We can probably use named tuples or [io.json_object](http://crystal-lang.org/api/JSON/Builder.html) to avoid allocating extra objects (should consume less memory and be a bit faster, though I don't know if it'll be noticeable)